### PR TITLE
chore: Update to clap v4

### PR DESCRIPTION
**Pull request by adamspofford-dfinity**
This updates clap to v4. Notable changes:

- Argument IDs are no longer case-converted in code; `foo_bar` means `conflicts_with("foo_bar")`, not `"foo-bar"`.
- 'Validators' have been replaced with parsers, to eliminate our habit of parsing twice.
- ArgAction replaces a number of special-cased functions like from_occurrences. 
- `#[clap]` has been replaced with `#[arg]` and `#[command]`.

The full high-level changelog can be found [here](https://epage.github.io/blog/2022/09/clap4/). 

This PR would have been delayed longer until `unstable-styles` is no longer unstable, but v3 blocks work by @smallstepman, so we should instead just be careful about updating clap in the lockfile. Also, these are more or less the default colors from v3, but we might want to look into using brand colors in certain places in a future PR.

### CODE REVIEW

File: src/dfx/src/commands/beta/project/import.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ pub struct ImportOpts {
     canister_name: Option<String>,
 
     /// Imports all canisters found in the other project.
-    #[clap(long, required_unless_present("canister-name"))]
+    #[clap(long, required_unless_present("canister_name"))]
```
Comment (smallstepman) [inline src/dfx/src/commands/beta/project/import.rs:19]:
```suggestion
    #[arg(long, required_unless_present("canister_name"))]
```

**Review by smallstepman — COMMENTED** [review]
shouldn't we use `arg` instead of `clap`? https://docs.rs/clap/latest/clap/index.html

**Comment by smallstepman** [comment]
since we're using `clap` as a workspace dependency, I think we should also update `icx-asset` (`#[clap` -> `#[arg/command`)

**Comment by adamspofford-dfinity** [comment]
It will be breaking in v5, but they're not emitting deprecation warnings for it.

**Review by ericswanson-dfinity — APPROVED** [review]

File: src/canisters/frontend/icx-asset/Cargo.toml [hunk 2]
```diff
@@ -12,9 +12,10 @@ categories = ["command-line-interface"]
 keywords = ["internet-computer", "agent", "icp", "dfinity", "asset"]
 
 [dependencies]
+anstyle.workspace = true
 anyhow.workspace = true
 candid = { workspace = true }
-clap = { workspace = true, features = ["derive", "cargo"] }
+clap = { workspace = true, features = ["derive", "cargo", "unstable-styles"] }
```
Comment (smallstepman) [inline src/canisters/frontend/icx-asset/Cargo.toml:18]:
```suggestion
clap = { workspace = true, features = ["derive", "cargo", "unstable-styles", "wrap_help" ] }
```

**Review by smallstepman — CHANGES_REQUESTED** [review]
looks like text wrapping for help command is behind feature flag now, you probably didn't notice on your widescreen ;) 
Screenshot 2023-05-04 at 17 30 03

**Review by smallstepman — APPROVED** [review]

**Pull request merged**